lib/honeycomb/client.rb

```diff
@@ -47,16 +48,7 @@ def initialize(configuration:)
     end
 
     def start_span(name:, serialized_trace: nil, **fields)
-      if context.current_trace.nil?
-        Trace.new(serialized_trace: serialized_trace,
-                  builder: libhoney.builder,
-                  context: context,
-                  **@additional_trace_options)
-      else
-        context.current_span.create_child
-      end
-
-      current_span = context.current_span
+      current_span = get_current_span(serialized_trace: serialized_trace)
```

Comment:
Rubocop said this method was too long now, so I just shuffled this "create a Trace if there isn't one, otherwise make a child from the current span" logic into a private method. Let me know if disabling Rubocop here would be preferable.